ludwig/models/llm.py

```diff
@@ -199,13 +189,28 @@ def __init__(
     def create_feature_dict(self) -> DictWrapper:
         return DictWrapper(LudwigFeatureDict())
 
-    def set_generation_config(self, generation_config_dict):
+    @contextlib.contextmanager
+    def use_generation_config(self, generation_config_dict: Optional[Dict[str, Any]] = None):
         """Sets the generation config for the model."""
-        self.generation = GenerationConfig(**generation_config_dict)
-
-    def reset_generation_config(self):
-        """Sets the generation config for th."""
-        self.generation = self.original_generation_config
+        # Save the original generation config so that we can reset it if/when we change it when self.generation gets is
+        # dynamically mutated during 1-off predict calls after fine-tuning.
+        original_generation_config_dict = self.generation.to_dict()
+        try:
+            # no-op if generation_config is None
+            if generation_config_dict is not None:
+                # unwrap the original generation config, update it with the new generation config
+                new_generation_config_dict = {**original_generation_config_dict, **generation_config_dict}
+                self._set_generation_config(new_generation_config_dict)
+            yield
+        finally:
+            self._set_generation_config(original_generation_config_dict)
+
+    def _set_generation_config(self, new_generation_config_dict: Dict[str, Any]):
+        self.generation = GenerationConfig(**new_generation_config_dict)
+        self.max_new_tokens = self.generation.max_new_tokens
+        # max input length value copied from FastChat
+        # https://github.com/lm-sys/FastChat/blob/0e958b852a14f4bef5f0e9d7a5e7373477329cf2/fastchat/serve/inference.py#L183  # noqa E501
+        self.max_input_length = self.context_len - self.max_new_tokens - 8
```

Comment:
Hi @geoffreyangus, it looks like https://github.com/ludwig-ai/ludwig/pull/3667 was merged. We might need to add this new LOC here as well:

`self.generation.pad_token_id = self.tokenizer.pad_token_id`

CC: @Infernaught @arnavgarg1

Reply:
got it, just added it in!